src/pal/src/misc/sysinfo.cpp

```diff
@@ -103,10 +103,18 @@ PAL_GetLogicalCpuCountFromOS()
     int nrcpus = 0;
 
 #if HAVE_SYSCONF
-    nrcpus = sysconf(_SC_NPROCESSORS_CONF);
+
+#if defined(_ARM_) || defined(_ARM64_)
+#define SYSCONF_GET_NUMPROCS       _SC_NPROCESSORS_CONF
+#define SYSCONF_GET_NUMPROCS_NAME "_SC_NPROCESSORS_CONF"
```

Comment:
A nit - instead of defining the macro and its string version, you can define a stringification macro that will make the conversion automatically:
```C
#define STRINGIFY1(s) #s
#define STRINGIFY(s) STRINGIFY1(s)
```
Then you can write the assert like this:
```C
ASSERT("sysconf failed for %s (%d)\n", STRINGIFY(SYSCONF_GET_NUMPROCS), errno); 
```

Reply:
Actually, I though about this once but I could not justify doing stringification instead of hard-coding names of the variables.

Unfortunately, doing stringification your way would not work for this example
```
#define _SC_NPROCESSORS_CONF 57
#define _SC_NPROCESSORS_ONLN 58

#define SYSCONF_GET_NUMPROCS       _SC_NPROCESSORS_CONF 
//#define SYSCONF_GET_NUMPROCS       _SC_NPROCESSORS_ONLN

#define STRINGIFY1(s) #s
#define STRINGIFY(s) STRINGIFY1(s)

int main()
{
    printf("STRINGIFY(SYSCONF_GET_NUMPROCS) is %s\n", STRINGIFY(SYSCONF_GET_NUMPROCS)); // Output: STRINGIFY(SYSCONF_GET_NUMPROCS) is 57
    return 0;
}
```
This is how SC_NPROCESSORS_CONF is defined in FreeBSD https://github.com/freebsd/freebsd/blob/06362ad468fc11671364e34ed413ea9c19d858f1/include/unistd.h#L292

Reply:
Ok, thanks, forget about it then.